build/common.sh

```diff
@@ -40,8 +40,8 @@ source "${KUBE_ROOT}/hack/lib/init.sh"
 readonly KUBE_BUILD_IMAGE_REPO=kube-build
 readonly KUBE_BUILD_IMAGE_CROSS_TAG="$(cat "${KUBE_ROOT}/build/build-image/cross/VERSION")"
 
-readonly KUBE_DOCKER_REGISTRY="${KUBE_DOCKER_REGISTRY:-k8s.gcr.io}"
-readonly KUBE_BASE_IMAGE_REGISTRY="${KUBE_BASE_IMAGE_REGISTRY:-k8s.gcr.io/build-image}"
+readonly KUBE_DOCKER_REGISTRY="${KUBE_DOCKER_REGISTRY:-registry.k8s.io}"
```

Comment:
when we build releases AFAIK we push directly to GCR, that still needs to push to k8s.gcr.io, since registry.k8s.io is read only.

i'm not sure if krel is pushing directly or pushing to some sort of staging first, but if it's not already overriding this it will need to start doing so.

Reply:
@puerco and @saschagrunert have confirmed this is ok now. (see https://github.com/kubernetes/release/pull/2522 in the note below from sasha and we updated @puerco on the sig-k8s-infra call)